miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/service/statsdb/TransformReport.java

```diff
@@ -0,0 +1,81 @@
+package uk.ac.bbsrc.tgac.miso.webapp.service.statsdb;
+
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.jdbc.core.JdbcTemplate;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+import uk.ac.bbsrc.tgac.miso.core.data.Run;
+import uk.ac.tgac.statsdb.run.ReportTable;
+import uk.ac.tgac.statsdb.run.Reports;
+import uk.ac.tgac.statsdb.run.ReportsDecorator;
+import uk.ac.tgac.statsdb.run.RunProperty;
+
+public abstract class TransformReport {
+
+  interface ReportRowConsumer {
+    public void consume(ObjectNode output, Map<String, ReportTable> reportResults, int index);
+  }
+  private static final Logger log = LoggerFactory.getLogger(TransformReport.class);
+
+  private String aName;
+
+  private final List<ReportRowConsumer> consumers = new ArrayList<>();
+
+  private final String metricType;
+
+  public TransformReport(String metricType) {
+    super();
+    this.metricType = metricType;
+  }
+
+  protected void add(String reportName, String property, int columnNumber) {
+    aName = reportName;
+    consumers.add(
+        (output, reportResults, index) -> output.put(property, reportResults.get(reportName).getTable().get(index).get(columnNumber)));
+  }
+
+  public void process(Run run, JdbcTemplate dataSource, ArrayNode output) {
+    Reports reports = new Reports(dataSource);
+    ReportsDecorator reportsDecorator = new ReportsDecorator(reports);
+    Map<RunProperty, String> params = new HashMap<>();
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Convert this Map to an EnumMap. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1640)